src/webapp/components/dataset-wizard/FilterIndicators.tsx

```diff
@@ -30,7 +30,13 @@ export type FilterIndicatorsProps = {
     theme: string;
     types: string[];
     selectedType: string;
-    hide: boolean;
+    hidden: boolean;
+    masterLogFrames: {
+        data: MasterLogFrame[];
+        error: Maybe<string>;
+        loading: boolean;
+        value: string;
+    };
```

Comment:
This kind of data structure benefits greatly from using discriminated union types, as it narrows down the expected states. We can create a generic type (and some supporting infrastructure) like this:

```typescript
type AsyncState =
  | { status: 'idle'; value: string }
  | { status: 'loading'; loading: true; value: string }
  | { status: 'success'; data: T[]; value: string }
  | { status: 'error'; error: string };
```

Only do this if you have time; otherwise, we can handle it this way for the next component that has an async loading state.